src/views/AnalysisModule/index.tsx

```diff
@@ -80,19 +81,35 @@ function AnalysisModule(props: AnalysisModuleProps) {
             notifyOnFailure(_ts('analysis', 'analysisModule'))({ error: errorBody }),
     });
 
+    const anaylsisObjectToEdit = useMemo(() => (
+        analyses?.find(a => a.id === analysisToEdit)
+    ), [analyses, analysisToEdit]);
+
+    const handleAnalysisEditClick = useCallback((analysisId) => {
+        setAnalysisToEdit(analysisId);
+        setModalShow();
+    }, [setModalShow]);
+
+    const handleNewAnalysisCreateClick = useCallback(() => {
+        setAnalysisToEdit(undefined);
+        setModalShow();
+    }, [setModalShow]);
+
     const analysisRendererParams = useCallback((key, data) => ({
+        analysisId: key,
+        onEdit: handleAnalysisEditClick,
         title: data.title,
         startDate: data.startDate,
         endDate: data.endDate,
-    }), []);
+    }), [handleAnalysisEditClick]);
 
     return (
         <div className={styles.analysisModule}>
             <SubNavbar>
                 <div className={styles.subNavbar}>
                     <Button
                         variant="primary"
-                        onClick={setModalShow}
+                        onClick={handleNewAnalysisCreateClick}
```

Comment:
Instead of this, isn't modalizing the button a better way to go?

Reply:
Yes, but we initially went with that approach because changing the states to view a modal was a pain. Now its pretty straight forward.